core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -153,6 +154,12 @@ void caseSensitive(boolean newCaseSensitive) {
   void delete(F file) {
     Preconditions.checkNotNull(file, "Cannot delete file: null");
     invalidateFilteredCache();
+    if (file.manifestLocation() != null) {
+      deletedManifestPositions
+          .computeIfAbsent(file.manifestLocation(), key -> Sets.newHashSet())
+          .add(file.pos());
```

Comment:
Note: The case where file.pos() is defined is the same as where file.manifestLocation() is defined; as long as the file is read from a manifest both of these will be set, so I don't think we need an additional null check for `pos`.

Reply:
I'm going to make this update and benchmark, I think it goes beyond memory usage we can avoid the unnecessary file path and partition tracking completely. It's basically unnecessary work that has a performance cost as well.

Reply:
Seems like validation currently is based on delete paths and also cherry picking logic for a reason I'm trying to figure out fails unless we track the delete paths here.

I'll take a look tomorrow to see how I can optimize this to avoid adding to delete paths